server/utils/functions.ts

```diff
@@ -0,0 +1,298 @@
+import {
+  last,
+  isObject,
+  isEmpty,
+  flatten,
+  find,
+  isString,
+  get,
+  isNil,
+  isArray,
+  first,
+} from 'lodash';
+import { Id, IStrapi, StrapiContentType, StrapiPlugin } from "strapi-typed";
+
+import { AuditLogContext, AuditLogParams, ContentTypeEntity, NavigationActions, NavigationItem, NavigationItemEntity, NestedPath, NestedStructure, PluginConfigNameFields, ToBeFixed } from "../../types";
+import { NavigationError } from '../../utils/NavigationError';
+import { TEMPLATE_DEFAULT, ALLOWED_CONTENT_TYPES, RESTRICTED_CONTENT_TYPES} from './constant';
+declare var strapi: IStrapi;
+
+export const getPluginService = <T>(name: string): T =>
+  strapi.plugin("navigation").service(name);
+
+export const errorHandler = (ctx: ToBeFixed) => (error: NavigationError | string) => {
+  if (error instanceof NavigationError) {
+    return ctx.badRequest(error.message, error.additionalInfo);
+  }
+  throw error;
+};
+
+export const parseParams = (params: ToBeFixed): any =>
+  Object.keys(params).reduce((prev, curr) => {
+    const value = params[curr];
+    const parsedValue = isNaN(Number(value)) ? value : parseInt(value, 10);
+    return {
+      ...prev,
+      [curr]: parsedValue,
+    };
+  }, {});
+
+
+
+export const templateNameFactory = async (
+  items: Array<NavigationItemEntity<ContentTypeEntity[] | ContentTypeEntity>> = [],
+  strapi: IStrapi,
+  contentTypes: Array<StrapiContentType<any>> = [],
```

Comment:
If we really need `any` please at least let's mark those places with `ToBeFixed`